src/premiscale/hypervisor/_base.py

```diff
@@ -69,4 +90,24 @@ def close(self) -> None:
         """
         Close the connection with the Libvirt hypervisor.
         """
-        self.connection.close()
\ No newline at end of file
+        if self._connection:
+            self._connection.virConnectClose()
+            log.info(f'Closed connection to host at {self.connection_string}')
+        else:
+            log.error(f'No host connection to close, probably due to an error on connection open.')
+
+    def _configure_ssh(self) -> None:
+        """
+        Configure the SSH connection to the host. This method makes connection timeouts configurable.
+        """
+        ssh_config_p = SSHConfig.from_path(os.path.expanduser('~/.ssh/config'))
+        ssh_config_entry = ssh_config_p.lookup(self._address_str)
+
+        if 'connecttimeout' not in ssh_config_entry:
+            with open(os.path.expanduser('~/.ssh/config'), 'a') as f:
+                if f'Host {self._address_str}' in f.read():
+                    raise ValueError(f'Host {self._address_str} already exists in ~/.ssh/config. Please remove it or add a ConnectTimeout value to it.')
+
+                f.write(f'\n\nHost {self._address_str}\n  ConnectTimeout {self.timeout}\n')
```

Comment:
This should be reworked at some point. It's a large dependency for not that much value-add.